stl/inc/xutility

```diff
@@ -2426,30 +2429,73 @@ namespace ranges {
     _EXPORT_STD template <range _Rng>
     using range_rvalue_reference_t = iter_rvalue_reference_t<iterator_t<_Rng>>;
 
+#if _HAS_CXX23
+    _EXPORT_STD template <class _Rng>
+    concept constant_range = input_range<_Rng> && _Constant_iterator<iterator_t<_Rng>>;
+
+    template <input_range _Rng>
+    _NODISCARD constexpr auto& _Possibly_const_range(_Rng& _Range) noexcept {
+        if constexpr (constant_range<const _Rng> && !constant_range<_Rng>) {
+#ifndef __clang__ // TRANSITION, DevCom-10187107
+            return _STD as_const(_Range);

```

Comment:
I would prefer we use `_STD as_const` always - `const_cast` should be kept to where it is _truly_ necessary.